lib/puppet/util/log.rb

```diff
@@ -170,6 +170,17 @@ def Log.with_destination(destination, &block)
   def Log.newmessage(msg)
     return if @levels.index(msg.level) < @loglevel
 
+    if msg.message.valid_encoding?
```

Comment:
I was on the fence about that... Since we're still logging the original message instead of a separate one re: the encoding I was hesitant to change too much about it - mostly to avoid surprising people and/or changing expectations/log levels in a (possibly?) backwards-incompatible way. I figured just logging at initial level was least surprising.